onnx/common/assertions.cc

```diff
@@ -24,7 +24,7 @@ std::string barf(const char* fmt, ...) {
   va_start(args, fmt);
 
   // use fixed length for buffer "msg" to avoid buffer overflow
-  vsnprintf(msg.data(), msg.size() - 1, fmt, args);
+  vsnprintf(static_cast<char*>(msg.data()), msg.size() - 1, fmt, args);
```

Comment:
args too?

warning: do not implicitly decay an array into a pointer; consider using gsl::array_view or an explicit cast instead [cppcoreguidelines-pro-bounds-array-to-pointer-decay] ```cpp vsnprintf(static_cast<char*>(msg.data()), msg.size() - 1, fmt, args);

Reply:
How (would we fix it for args)? This is used via a macro for printf style untyped inputs.

Reply:
Not sure. Just saw the lint message. Please feel free to ignore.